### PR TITLE
Fixed deprecated code in ANL

### PR DESCRIPTION
**Pull request by Robertdebrus**
Converted all the worker options to the new format in 1.13.2a.
I also changed any tabs in the files to four spaces.

I have fixed the code (with spacing), but I am wondering how to get rid of the old commits.

I hope I did everything correctly, because I am very new at this.
My Wesnoth forum name is Robertdebrus

### CODE REVIEW

**Comment by Vultraz** [comment]
Looks good, but you forgot the four-space indent before each of the new lines. Also, could you possibly do the same change for any other occurrences of this syntax in ANL?

**Comment by Robertdebrus** [comment]
Oh, thanks for reminding me about the spacing, and I will try and do the rest!

**Comment by Vultraz** [comment]
The spacing still seems to have really gotten off (tabs vs spaces) in some places. Perhaps you would consider running the 'wmlindent' scrip in data/tools on the directory - it automatically applies correct spacing - and then rebasing.

Besides that, looks fine to me :)

**Comment by Robertdebrus** [comment]
I will try wmlindent!

**Comment by Robertdebrus** [comment]
It gives me error code 1 and says

`Traceback (most recent call last):

  File "C:\Users\***\Documents\GitHub\wesnoth\data\tools\wmlindent", line 310, in <module>

    arguments, "|".join(exclude))

  File "C:\Users\***\Documents\GitHub\wesnoth\data\tools\wmlindent", line 239, in convertor

    for filename in allwmlfiles(arg):

  File "C:\Users\***\Documents\GitHub\wesnoth\data\tools\wmlindent", line 226, in allwmlfiles

    if wmltools.vcdir in dirs:

AttributeError: 'module' object has no attribute 'vcdir'
`

**Comment by Robertdebrus** [comment]
Never mind, It's running now

**Comment by Robertdebrus** [comment]
Or not

**Comment by Robertdebrus** [comment]
Travis is saying that one of the files has an error or something, but if I temporarily put the files in the game, it still works, and has no errors, or anything

**Comment by Robertdebrus** [comment]
I am new to this, so I am confused as to rebasing

**Comment by gfgtdf** [comment]
This still has wrong spacing in a lot of files.

**Comment by Robertdebrus** [comment]
Yeah, I seems to be making a mess of things. I think I clicked the "commit" button too often, and I am still trying to sort things out... I will look for the spacing issues.

**Comment by Robertdebrus** [comment]
I have gotten wmlindent to work (wrong python), but there are 16 different commit that I am trying to join together, but seem to just make more, so I am afraid to try and put the updated files on.

Help?

**Comment by Robertdebrus** [comment]
Is there anything else?

File: data/multiplayer/scenarios/ANL_utils/ANL_leader_options.cfg [hunk 1]
```diff
@@ -369,7 +374,7 @@ Help an ally with their research")}
                                         [message]
                                             caption= _ "Diplomatic Options"
                                             speaker=unit
-                                            image=items/book4.png
+                                            image= "items/book4.png"
```
Comment (CelticMinstrel) [inline data/multiplayer/scenarios/ANL_utils/ANL_leader_options.cfg]: This change is pointless.

Comment (Robertdebrus) [inline data/multiplayer/scenarios/ANL_utils/ANL_leader_options.cfg]: Oh, sorry, I had seen it like that earlier, so I thought it was something I had messed up accidentally.
Does that mean I should change it back?

File: data/multiplayer/scenarios/ANL_utils/ANL_research_options.cfg [hunk 2]
```diff
@@ -51,31 +51,31 @@
...
-#textdomain wesnoth-anl
-        [/message]
-
-        [set_variable]
-            name=player_$side_number|.warfare.troop_available
-            sub=1
-        [/set_variable]
-    [/command]
-# wmlindent: stop ignoring
+    # wmlindent: start ignoring
```
Comment (CelticMinstrel) [inline data/multiplayer/scenarios/ANL_utils/ANL_research_options.cfg]: This chunk of changes is whitespace-only and thus meaningless.

File: data/multiplayer/scenarios/ANL_utils/ANL_research_options.cfg [hunk 3]
```diff
@@ -131,9 +131,9 @@
                         # wmllint: display on
                         message= _ "We are currently studying $player_$side_number|.research.target_language_name|. To which end would you have our scholars devote their minds?
 
-Our farms produce $player_$side_number|.farming.gold|g
-Our mines produce $player_$side_number|.mining.gold|g
-"
+    Our farms produce $player_$side_number|.farming.gold|g
+    Our mines produce $player_$side_number|.mining.gold|g
+    "
```
Comment (CelticMinstrel) [inline data/multiplayer/scenarios/ANL_utils/ANL_research_options.cfg]: This is also whitespace-only and probably _not_ meaningless, which is bad.

File: data/multiplayer/scenarios/ANL_utils/ANL_research_options.cfg [hunk 4]
```diff
@@ -154,9 +154,10 @@ Our mines produce $player_$side_number|.mining.gold|g
...
 
                         [option]
-                            message= {MENU_IMG_TXT "items/flower4.png" ( _ "<span color='green'>Agriculture</span>
-Farmers produce +1 gold
-Study Progress: $player_$side_number|.farming.progress|/$player_$side_number|.farming.target")}
+                            image = "items/flower4.png"
+                            message = _ "<span color='green'>Agriculture</span>
+    Farmers produce +1 gold
+    Study Progress: $player_$side_number|.farming.progress|/$player_$side_number|.farming.target"
                             [command]
```
Comment (CelticMinstrel) [inline data/multiplayer/scenarios/ANL_utils/ANL_research_options.cfg]: Again you've added extra whitespace within a string. Please don't do this. There are also two more cases of it in this file.

File: data/multiplayer/scenarios/ANL_utils/ANL_leader_options.cfg (hunk 1 above)
Comment (CelticMinstrel) [inline data/multiplayer/scenarios/ANL_utils/ANL_leader_options.cfg]: I would, but it _is_ just a single line, so it's not hugely important. More important are all the whitespace problems.

File: data/multiplayer/scenarios/ANL_utils/ANL_research_options.cfg (hunk 4 above)
Comment (Robertdebrus) [inline data/multiplayer/scenarios/ANL_utils/ANL_research_options.cfg]: Hmm... I don't recall putting that there, or anywhere, probably it was wmlindent, or an accidental whitespace when I did not notice...

**Comment by CelticMinstrel** [comment]
Ultimately, we would like to be able to click the "Files Changed" tab on this page and see a nice summary of what you've done. If you change whitespace all over the place, that gets in the way of the actually stuff you've changed, making it difficult to impossible to know what you actually did. This unfortunately may mean that running wmlindent actually made the situation worse.

The other issue with this PR is the sheer number of commits that each affect only a single file. Making small commits isn't a bad thing, but keeping related changes in multiple files together is a good thing. If you can figure out how to rebase and squash the related commits together, that would be great (but best to hold off until it's approved first). If not, though, it wouldn't be that hard for me to do it myself; just keep it in mind in case of future contributions.

File: data/multiplayer/scenarios/ANL_utils/ANL_research_options.cfg (hunk 4 above)
Comment (CelticMinstrel) [inline data/multiplayer/scenarios/ANL_utils/ANL_research_options.cfg]: Yeah, I too suspect it might have been wmlindent.

**Comment by aginor** [comment]
Seeing as how nobody has done this yet, I'd like to invite you to join us on our chat channel. You can find it on https://webchat.freenode.net/?channels=wesnoth-dev

That way we can help you sort out any practical issues in real-time, as well as talk you through rebasing when it comes to that :)
Talking on IRC isn't a substitute for going through the review of a pull-request, but we can certainly advice you how to address some of the feedback :)

**Comment by Robertdebrus** [comment]
CelticMinistrel, the rebasing would be a great help, because I seem to not be experienced enough with git to do so, and indeed, several of those problems you mentioned were probably caused by my attempts to rebase.

aginor, thanks for the invite!

**Comment by gfgtdf** [comment]
There are still a lot of indents changed. The main problems eems tobe that you also changed a lot indents inside a 'wmlindent: start ignoring' block which wmlindent cnnot fix for you (because it ignores these blocks)

**Comment by Robertdebrus** [comment]
Oh, right, forgot about those!

**Comment by Robertdebrus** [comment]
This looks like it fixes the spacing problems.

Sorry or taking so long

**Comment by gfgtdf** [comment]
in ANL_research_options.cfg lines 54 and 136 are still shiwspace issues.

**Comment by Robertdebrus** [comment]
Got it.

**Comment by gfgtdf** [comment]
Ok the whitespace issues see to be fixed now.
What you still need to do it to merge all these commit into two (one for the about.cfg chagne and one for the rest)

To do this you need to use the gut commandline use "git rebase HEAD~21 -i" move the line about the about cfg change commit to the top and then mark all fixup commits as fixup.

**Comment by Vultraz** [comment]
In ANL_worker_options.cfg, all but one of the changed blocks is missing 4-space indent. Beside that, looks good.

**Comment by gfgtdf** [comment]
hm yes but afaik this was also the case before this patch to this is not really an issue.

**Comment by Robertdebrus** [comment]
I have and am having great difficulties rebasing...

**Comment by CelticMinstrel** [comment]
For this time, I took care of it for you. Your trouble came from those two random merge commits - in future, I suggest using "git pull --ff-only" when pulling in order to avoid unexpected merges.

If you want to make further contributions, you will need to reset your repository first. The easiest way to do this at the command-line is probably "git reset --hard HEAD~30" followed by pulling from the main Wesnoth repository. Then, "git push --force" to update your github fork.

**Pull request closed without merging**